#7984 extending geoserver integration documentation (#8104)
geosolutions-it/MapStore2: docs/developer-guide/integrations/users/geoserver.md

> # GeoServer integrations
> 
> With the integration of MapStore you can create a fine grained access to the data and the services for the users
> 
> ## MapStore2 users  with Authkey

Reviewer: ```suggestion
# GeoServer integrations

## MapStore/GeoServer users integration

MapStore can share users, groups an roles with GeoServer. This type of integration allows to setup a fine grained access to the data and the services based on MapStore groups and roles.
```

 - Verify that "useAuthenticationRules" is set to `true`
 - `authenticationRules` array should contain 2 rules:
     - The first rule should already be present, and defines the authentication method used internally in mapstore
     - The second rule (the one you need to add) should be added and defines how to authenticate to GeoServer:
         - `urlPattern`: is a regular expression that identifies the request url where to apply the rule
         - `method`: set it to `authkey` to use the authentication filter you just created in Geoserver.
         - `authkeyParamName`: is the name of the authkey parameter defined in GeoServer (set to `authkey` by default)

> ## Other geoserver integration
> 
> You can allow some users to:

Reviewer: ```suggestion
### Advantages of user integration

Integrating the user/groups database with GeoServer you can allow some users to:
```

- Execute some processes (via [WPS security](http://gs-stable.geo-solutions.it/geoserver/web/wicket/bookmarkable/org.geoserver.wps.web.WPSAccessRulePage?25&filter=false)) 

>   For example download feature can leverage gs:download process to allow user to download data

Reviewer: ```suggestion
- Setup the download extrension to allow/deny certain users to download data
```

> - Edit Styles (by default allowed only to administrators, but you can change it acting on /rest/ Filter Chains). 

Reviewer: ```suggestion
- Edit Styles (by default allowed only to administrators, but you can change it acting on `/rest/` Filter Chains). 
```

> - Access to layers based on users (standard geoserver security)

Reviewer: ```suggestion
- Access to layers based on users (using the standard GeoServer security)
```

- Filter layers data based on users (GeoFence), see [here](https://docs.geoserver.org/latest/en/user/extensions/geofence-server/index.html)

- Allow editing of layers to certain MapStore users ([GeoServer Security](https://docs.geoserver.org/stable/en/user/security/webadmin/data.html)). The editing can be enabled in the [plugin settings of MapStore](https://dev-mapstore.geosolutionsgroup.com/mapstore/docs/api/plugins#plugins.FeatureEditor)

## GeoServer Plugins and Extensions

> MapStore supports several plugins for GeoServer

Reviewer: ```suggestion
MapStore supports several plugins for GeoServer. Installing them will expand the functionalities of MapStore, allowing to navigate data with time dimension, styling layers and so on.

Here a list of the extensions that MapStore can use: 
```

> - [WMTS Multidimensional](https://docs.geoserver.org/stable/en/user/community/wmts-multidimensional/index.html) despite the name, this service provides multidimensional discovery services for GeoServer in general, and is **required** to use the timeline plugin of MapStore.

Reviewer: ```suggestion
- [WMTS Multidimensional](https://docs.geoserver.org/stable/en/user/community/wmts-multidimensional/index.html) despite the name, this service provides multidimensional discovery services for GeoServer in general, not only for WMTS, and it is **required** to use the timeline plugin of MapStore.
```

> - [SLD Rest Service](https://docs.geoserver.org/latest/en/user/extensions/sldservice/index.html) Is an extension that can be used by the MapStore styler to classify Vector and Raster data. It can inspect the real layer data to apply classification based on values contained in it. It allows to select various classification types (quantile, equalInterval, standardDeviation…) and to customize the color scales based on parameters

Reviewer: ```suggestion
- [SLD Rest Service](https://docs.geoserver.org/latest/en/user/extensions/sldservice/index.html) is an extension that can be used by the MapStore styler to classify Vector and Raster data. It can inspect the real layer data to apply classification based on values contained in it. It allows to select various classification types (quantile, equalInterval, standardDeviation…) and to customize the color scales based on parameters
```

> - [CSS Extension](https://docs.geoserver.org/latest/en/user/styling/css/install.html): With this extension MapStore can edit styles in CSS format, in addition to the standard SLD format

Reviewer: ```suggestion
- [CSS Extension](https://docs.geoserver.org/latest/en/user/styling/css/install.html): With this extension the MapStore styler allows to edit styles also in CSS format, in addition to the standard SLD format
```
